src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/LetterSentOrRecievedHandlerTest.java

```diff
@@ -66,6 +67,42 @@ void handling_should_throw_if_cannot_actually_handle() {
                 .isExactlyInstanceOf(IllegalStateException.class);
     }
 
+    @Test
+    void should_write_sent_in_letter_sent_or_received() {
+
+        when(asylumCase.read(IS_OUT_OF_COUNTRY_ENABLED, YesOrNo.class)).thenReturn(Optional.of(isOutOfCountryEnabled));
+        when(asylumCase.read(APPELLANT_IN_UK, YesOrNo.class)).thenReturn(Optional.of(appellantInUk));
+
+        when(asylumCase.read(IS_ACCELERATED_DETAINED_APPEAL, YesOrNo.class)).thenReturn(Optional.of(isAcc));
+        when(asylumCase.read(APPELLANT_IN_DETENTION, YesOrNo.class)).thenReturn(Optional.of(appellantInDetention));
+
+        PreSubmitCallbackResponse<AsylumCase> callbackResponse =
+                letterSentOrRecievedHandler.handle(PreSubmitCallbackStage.MID_EVENT, callback);
+
+        assertNotNull(callbackResponse);
+        assertEquals(asylumCase, callbackResponse.getData());
+        verify(asylumCase, times(1)).write(LETTER_SENT_OR_RECEIVED, "Received");
+
+    }
+
+    @Test
+    void should_write_sent_in_letter_sent_or_sent() {
```

Comment:
Can you please correct the names of the unit tests:
should_write_received_for_appellantInUk_accelerateDetainedAppeal() &
should_write_sent_for_non_acceleratedDetainedAppeal()